testsuite/features/step_definitions/navigation_steps.rb

```diff
@@ -461,6 +466,14 @@
   end
 end
 
+Then(/^the table row for "([^"]*)" should( not)? contain "([^"]*)" icon$/) do |row, should_not, icon_class|
+  xpath_query = "//div[@class=\"table-responsive\"]/table/tbody/tr[.//*[contains(.,'#{row}')]]"
+  within(:xpath, xpath_query) do
+    func = should_not ? :has_no_css? : :has_css?
```

Comment:
Nice code, but harder to debug if it fails.
The stacktrace will have an extra jump.
I would have an if statement for should_not variable, calling has_css or has_no_css

Reply:
I re-wrote this according to Tino in the end. Rubocop was also complaining about this (`send(..`).